sql/parser/token.go

```diff
@@ -354,12 +360,14 @@ func (tkn *tokenizer) scanNumber(seenDecimalPoint bool) (int, []byte) {
 
 fraction:
```

Comment:
It predates this commit, but octal floats? Shudder. And octal floats have base 10 fractional components and exponents? I can't imagine a case where this would be anything but a bug magnet. Unless there is a very strong reason I'd rather say that bases other than 10 are only allowed for integers. And for octal, if we support it at all (is there any good reason for using octal besides `chmod`?), I'd rather use a more explicit prefix than the traditional leading zero (i.e. python 3's `0o`, by way of analogy with `0x`. This also generalizes nicely to binary `0b`, which I would find much more palatable than octal).